Dockerfile

```diff
@@ -15,19 +35,21 @@ RUN conda create -y --name curator -c conda-forge -c nvidia \
   libcublas \
   libcurand \
   libcusparse \
-  libcusolver
+  libcusolver && \
+  source activate curator && \
+  pip install --upgrade cython pytest pip
 
-# Needed to navigate to and pull the forked repository's changes
-ARG FORKED_REPO_URL
-ARG CURATOR_COMMIT
+RUN \
+  --mount=type=bind,source=/opt/NeMo-Curator/nemo_curator/__init__.py,target=nemo_curator/__init__.py,from=curator-update \
+  --mount=type=bind,source=/opt/NeMo-Curator/pyproject.toml,target=pyproject.toml,from=curator-update \
+  source activate curator && \
+  pip install ".[all]"
+
+COPY --from=curator-update /opt/NeMo-Curator/ /opt/NeMo-Curator/
```

Comment:
Ah okay, I see. By splitting the repository logic and the installation logic into 2 separate stages, it is allowing us to take advantage of caching in the environment setup portion (for example, the libcu* installations)?

Reply:
Exactly! This layer will only be build if the files `__init__.py` or `pyproject.toml` change. Otherwise it'll use the cached instance